naga/src/back/mod.rs

```diff
@@ -22,6 +22,13 @@ const BAKE_PREFIX: &str = "_e";
 
 type NeedBakeExpressions = crate::FastHashSet<crate::Handle<crate::Expression>>;
 
+/// Specifies the values of pipeline-overridable constants in the shader module.
+///
+/// The key represents the numeric ID of the constant, if one is specified, and otherwise the constant's identifier name.
```

Comment:
Suggest:

> The key is the constant's identifier name, or the pipeline constant ID as an decimal ASCII number, if an `@id` attribute was specified.

Reply:
I think we should make it clearer that it must be the id if one was specified.

How about:
```
If an `@id` attribute was specified on the declaration,
the key must be the pipeline constant ID as a decimal ASCII number; if not,
the key must be the constant's identifier name.
```

Reply:
Yes, even better.